libs/api/domains/regulations/src/lib/dto/getRegulation.input.ts

```diff
@@ -0,0 +1,26 @@
+import { ISODate, RegName } from '@island.is/clients/regulations'
+import { Field, InputType } from '@nestjs/graphql'
+import { IsOptional, IsString } from 'class-validator'
+
+@InputType()
+export class GetRegulationInput {
+  @Field()
+  @IsString()
+  viewType!: 'current' | 'diff' | 'original' | 'd'
```

Comment:
Having these as enums would provide better developer experience in playground and provide more detailed error responses when input does not match intended values.

<img width="1703" alt="Screenshot 2021-04-29 at 15 05 57" src="https://user-images.githubusercontent.com/6640435/116573598-7a2e1f00-a8fc-11eb-978a-9500143ef8c0.png">

Reply:
ok. I'm generally not a fan of enums ([neither are the TypeScript developers themselves](https://youtu.be/8qm49TyMUPI?t=542) as it heppens) but I see how it provides value because of how the playground works.

Would using a `const enum` be ok?

Reply:
Fixed

Reply:
Consider me converted to the anti enum train @maranomynet.
I wasn't sure about this I ran a few tests `const enums` don't seem to work nestjs/class-valdiator probably rely on enums staying enumy.
The only way I got them to behave as advertised was with something like:
```ts
enum MegaNice {
  Stuff = 'stuff',
  Other = 'other'
}

registerEnumType(MegaNice, {
  name: 'MegaNice',
})

@InputType()
export class CreateMegaNiceDto {
  @Field(() => MegaNice)
  @IsEnum(MegaNice)
  megaNiceStuff!: MegaNice
}
```

Reply:
Eek! This seems a whole lot of faffing, compared to

```
  viewType!: 'current' | 'diff' | 'original' | 'd'
```

Reply:
@Valur ☝️

Reply:
ok ég hugsa að þetta sé komið núna